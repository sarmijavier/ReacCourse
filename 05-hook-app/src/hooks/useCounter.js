import { useState } from 'react'

export const useCounter = ( initialState = 10) => {
    //f2 to change all names
    const [counter, setCounter] = useState(initialState)    

    const reset = () => {
        setCounter( initialState )
    }

    const increment = () => {
        setCounter( counter + 1)
    }

    const decrement = () => {
        setCounter( counter - 0)
    }

    return {
        counter,
        increment,
        decrement,
        reset
    }
}
