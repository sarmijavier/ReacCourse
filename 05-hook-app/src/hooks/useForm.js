import { useState } from 'react'


export const useForm = ( initialState= {} ) => {
    // here we can do validations, or extra procces
    const [values, setValues] = useState(initialState)
    const handleInputChange = ({ target }) => {

        setValues({
            ...values,
            [ target.name ]: target.value

        })
    }

    return [ values, handleInputChange ]
}