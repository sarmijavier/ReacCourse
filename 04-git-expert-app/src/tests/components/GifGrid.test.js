import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import React from 'react';
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');



describe('Pruebas con GifGrid component', () => {
    const category = 'Dragon ball z';
    
    test('Deber√≠a de mostrar <GifGrid /> correctamente', () => {
        

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });


        const wrapper = shallow( <GifGrid category={ category }/>);
        expect( wrapper ).toMatchSnapshot();

    });

    test('debe de mostrar items cuando se cargan imagines useFetchGifs', () => {
    
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title:'Cualquier cosa'
        },
    
        {
            id: '123',
            url: 'https://localhost/cualquier/cosa.jpg',
            title:'Cualquier cosa'
        }
        ];
        

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false 
        });
        const wrapper = shallow( <GifGrid category={ category }/>);
    
        expect( wrapper ).toMatchSnapshot
        expect( wrapper.find('p').exists() ).toBe(false)
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length )
    })
    
    


})