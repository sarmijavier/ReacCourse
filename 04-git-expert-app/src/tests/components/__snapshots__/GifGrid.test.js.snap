// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con GifGrid component Deber√≠a de mostrar <GifGrid /> correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInLeftBig"
  >
    Dragon ball z
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
